tests/test_main.py

```diff
@@ -616,33 +1006,168 @@ def check_code(self, code, nodes, decorators, check_names):
+                    if not py11 and inst.opname == 'STORE_ATTR' and inst.argval in ['sign', '__annotations__']:
                         continue
 
                     if inst.opname == 'LOAD_FAST' and inst.argval == '.0':
                         continue
 
-                    if inst.argval == 'AssertionError':
+                    if inst.argval == "AssertionError":
                         continue
 
                     if any(
                         isinstance(stmt, (ast.AugAssign, ast.Import))
                         for stmt in source.statements_at_line(lineno)
                     ):
                         continue
+
+                    # report more information for debugging
+                    print(e)
+                    print("search bytecode", inst)
+                    print("in file", source.filename)
+
+                    if sys.version_info >= (3,11):
+                        print("at position", inst.positions)
+
+                        with open(source.filename) as sourcefile:
+                            source_code = sourcefile.read()
+
+                        dump_source(source_code, start(inst), end(inst))
+
+                        options = []
+                        for node in ast.walk(ast.parse(source_code)):
+                            if not hasattr(node, "lineno"):
+                                continue
+
+                            # if {start(node), end(node)} & {start(inst), end(inst)}:
+                            if start(node) <= end(inst) and start(inst) <= end(node):
+                                options.append(node)
+
+                        for node in sorted(
+                            options,
+                            key=lambda node: (
+                                node.end_lineno - node.lineno,
+                                node.end_col_offset - node.col_offset,
+                            ),
+                        ):
+
+                            if (node.end_lineno - node.lineno) > (
+                                inst.positions.end_lineno - inst.positions.lineno
+                            ) * 4:
+                                # skip nodes which are way to long
+                                continue
+
+                            print()
+                            print(
+                                "possible node",
+                                node.lineno,
+                                node.end_lineno,
+                                node.col_offset,
+                                node.end_col_offset,
+                                ast.dump(node),
+                            )
+
                     raise
                 # argval isn't set for all relevant instructions in python 2
-                if isinstance(node, ast.Name) and (PY3 or inst.argval):
-                    self.assertEqual(inst.argval, node.id)
+                if isinstance(node, ast.Name) and (PY3 or inst.argval) and False:
```

Comment:
and False?

Reply:
sry, missed this one. I skip it now only in 3.11. The assert does not hold any more because of all the new bytecodes name mangleling and so on.